postgres/datadog_checks/postgres/postgres.py

```diff
@@ -907,75 +907,77 @@ def _get_custom_queries(self, db, tags, custom_queries, programming_error):
-                                "metric_prefix `{}`".format(column_type, name, metric_prefix)
+                                "column field `name` is required for metric_prefix `{}`".format(metric_prefix)
                             )
                             break
-                        try:
-                            metric_info.append((
-                                '{}.{}'.format(metric_prefix, name),
-                                float(value),
-                                column_type
-                            ))
-                        except (ValueError, TypeError):
+
+                        column_type = column.get('type')
+                        if not column_type:
                             self.log.error(
-                                "non-numeric value `{}` for metric column `{}` of "
-                                "metric_prefix `{}`".format(value, name, metric_prefix)
+                                "column field `type` is required for column `{}` "
+                                "of metric_prefix `{}`".format(name, metric_prefix)
                             )
                             break
 
-                # Only submit metrics if there were absolutely no errors - all or nothing.
-                else:
-                    for info in metric_info:
-                        metric, value, method = info
-                        getattr(self, method)(metric, value, tags=query_tags)
+                        if column_type == 'tag':
+                            query_tags.append('{}:{}'.format(name, value))
+                        else:
+                            if not hasattr(self, column_type):
+                                self.log.error(
+                                    "invalid submission method `{}` for column `{}` of "
+                                    "metric_prefix `{}`".format(column_type, name, metric_prefix)
+                                )
+                                break
+                            try:
+                                metric_info.append((
+                                    '{}.{}'.format(metric_prefix, name),
+                                    float(value),
+                                    column_type
+                                ))
+                            except (ValueError, TypeError):
+                                self.log.error(
+                                    "non-numeric value `{}` for metric column `{}` of "
+                                    "metric_prefix `{}`".format(value, name, metric_prefix)
+                                )
+                                break
+
+                    # Only submit metrics if there were absolutely no errors - all or nothing.
+                    else:
+                        for info in metric_info:
+                            metric, value, method = info
+                            getattr(self, method)(metric, value, tags=query_tags)
```

Comment:
We should call out in the readme (or perhaps detect it and log a message/emit only the first one?) that there could be issues if your query doesn't contain a `tag` element that is unique between the returned rows. 

If the tag is non unique between rows that are returned, for gauges this means that only the last value would be emitted, and it may mess up the calculations for rates.

Reply:
Can you please explain a bit more?

Reply:
Yeah. So if the column they're using as a tag is something like "First Name" and it's not exactly unique, they may get two results for a query with `Nick` as the First Name column. Then we'd be sending the same metric with the same tag. For gauges, only the last would be emitted and for gauges, it may not play nicely with the delta calculation.